packages/shopify-ui-extensions/src/react/context.ts

```diff
@@ -0,0 +1,16 @@
+import type { AnyClient } from "@gadgetinc/api-client-core";
+import { createContext, useContext } from "react";
+
+export type GadgetShopifyUiExtensionContextValue = {
+  /** Whether or not the api client is ready to make requests to with Shopify session token authentication */
+  ready: boolean;
+  /** An instance of the Gadget application's api client set up to use Shopify session token authentication */
+  api: AnyClient;
+};
+
+export const GadgetShopifyUiExtensionContext = createContext<GadgetShopifyUiExtensionContextValue>({
+  ready: false,
+  api: null as unknown as AnyClient,
+});
+
+export const useGadget = () => useContext<GadgetShopifyUiExtensionContextValue>(GadgetShopifyUiExtensionContext);
```

Comment:
I think it might be good to make this generic so that folks can do

```typescript
useGadget<MyClient>().api // => MyClient
```

```suggestion
export type GadgetShopifyUIExtensionContextValue<Client extends AnyClient = AnyClient> = {
  /** Whether or not the api client is ready to make requests to with Shopify session token authentication */
  ready: boolean;
  /** An instance of your Gadget app's api client, which has been set up to use Shopify session token authentication */
  api: Client;
};

export const GadgetShopifyUIExtensionContext = createContext<GadgetShopifyUIExtensionContextValue>({
  ready: false,
  api: null as unknown as AnyClient,
});

export const useGadget = <Client extends AnyClient = AnyClient>() => useContext<GadgetShopifyUIExtensionContextValue<Client>>(GadgetShopifyUIExtensionContext);
```

Reply:
:+1: